src/backend/auth/jwt_utils.py

```
@@ -0,0 +1,105 @@
+def require_user(
+    credentials: HTTPAuthorizationCredentials | None = Depends(_bearer),
+) -> dict | None:
+    """Enforced only when settings.AUTH_ENFORCED; otherwise permissive."""
+    user = _decode_to_user(credentials)
+    if settings.AUTH_ENFORCED and user is None:
+        raise HTTPException(401, "Not authenticated", headers=_UNAUTH_HEADERS)
+    return user
+
+
+def require_admin(
+    credentials: HTTPAuthorizationCredentials | None = Depends(_bearer),
```

Ruff (0.15.15)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)